Benchmarks/GaussianElimination.cs

```diff
@@ -0,0 +1,278 @@
+                    }
+
+                    // Swap pivot row with the current row.
+                    if (pi != j)
+                    {
+                        for (int n = 0; n <= N; n++)
+                        {
+                            var tmp = Ab[n + pi * eN];
+                            Ab[n + pi * eN] = Ab[n + j * eN];
+                            Ab[n + j * eN] = tmp;
+                        }
+                    }
+
+                    // Eliminate the rows after the pivot.
+                    double p = Ab[j * eN + j];
+                    for (int i = j + 1; i < M; ++i)
+                    {
+                        double s = Ab[i * eN + j] / p;
+                        if (s != 0.0)
+                            for (int ij = j + 1; ij <= N; ++ij)
+                                Ab[i * eN + ij] -= Ab[j * eN + ij] * s;
+                    }
+                }
+            }
+        }
+
+        [Benchmark(Baseline = true)]
+        public void ArrayOfArrays()
+        {
+            for (int iteration = 0; iteration < Size; iteration++)
+            {
+                var Ab = _data[iteration].JaggedArray;
+
+                // Solve for dx.
+                // For each variAb2le in the system...
+                for (int j = 0; j + 1 < N; ++j)
+                {
+                    int pi = j;
+                    double max = Math.Abs(Ab[j][j]);
+
+                    // Find a pivot row for this variable.
+                    for (int i = j + 1; i < M; ++i)
+                    {
+                        // if(|JxF[i][j]| > max) { pi = i, max = |JxF[i][j]| }
+                        double maxj = Math.Abs(Ab[i][j]);
+                        if (maxj > max)
+                        {
+                            pi = i;
+                            max = maxj;
+                        }
+                    }
+
+                    // Swap pivot row with the current row.
+                    if (pi != j)
+                    {
+                        var tmp = Ab[pi];
+                        Ab[pi] = Ab[j];
+                        Ab[j] = tmp;
+                    }
+
+                    // Eliminate the rows after the pivot.
+                    double p = Ab[j][j];
+                    for (int i = j + 1; i < M; ++i)
+                    {
+                        double s = Ab[i][j] / p;
+                        if (s != 0.0d)
+                        {
+                            for (int jj = j + 1; jj <= N; ++jj)
+                            {
+                                Ab[i][jj] -= Ab[j][jj] * s;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        public void ArrayOfArraysVectorized()
+        {
+            for (int iteration = 0; iteration < Size; iteration++)
+            {
+                var Ab = _data[iteration].JaggedArray;
+
+                const double tiny = 0.00001;
```

Comment:
The other tests compare with exactly 0, maybe this could affect the timing of the results?

Also, in practice, it would probably be necessary to make this a function of the pivot value, e.g. `maxj * 1e-6`. It's hard to come up with an 'epsilon' that always works without some understanding of the magnitude of the values being handled.

Reply:
Switched to '0.0' for now - at least we are comparing apples to apples.